### PR TITLE
Expand cocina-models documentation to make release process clearer

### LINKED ISSUE #247: Clarify README about orchestration of changes when models change
I.e.:

1. Update cocina-models
2. Update clients
3. Update dsa & sdr-api
4. Update other codebases as necessary depending on whether bw-compat or not, etc.

### PR DESCRIPTION
**Pull request by mjgiarlo**
Fixes #247

## Why was this change made?

Increased clarity for the team

## How was this change tested?

N/A

## Which documentation and/or configurations were updated?

README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,44 +34,53 @@ Beyond what is necessary to test the generator, the Cocina model classes are not
...
+
+### Step 1: Cut the release
+
 The release process is much like any other gem. First bump the version in `lib/cocina/models/version.rb`, and commit the result. Then run:
 ```
 bundle exec rake release
 ```
 which pushes the gem to rubygems.org.  Next write up the release notes: https://github.com/sul-dlss/cocina-models/releases .
 
-Finally, you must release versions of [sdr-client](https://github.com/sul-dlss/sdr-client) and [dor-services-client](https://github.com/sul-dlss/dor-services-client/) pinned to this version because [Argo](https://github.com/sul-dlss/argo) depends on both of those. When [dor-services-app](https://github.com/sul-dlss/dor-services-app) is updated to use the new models (via the auto-update script), the clients must be updated at the same time or there is risk of models produced by dor-services-app not being acceptable to the clients.
```
Comment (mjgiarlo) [inline README.md:43]: None of the text in the README was actually removed. It was just shifted around a bit to hopefully communicate the organization and order of information more clearly and more effectively.

File: README.md [hunk 2]
```diff
@@ -34,44 +34,53 @@ Beyond what is necessary to test the generator, the Cocina model classes are not
...
+### Step 2: Update client gems coupled to the models
 
-### Communicate
+Next, you should release versions of [sdr-client](https://github.com/sul-dlss/sdr-client) and [dor-services-client](https://github.com/sul-dlss/dor-services-client/) pinned to this version because applications such as [Argo](https://github.com/sul-dlss/argo) depend on both of these gems using the same models.
```
Comment (ndushay) [inline README.md:51]: do you want to say anything about needing to do this before updating APIs due to roundtrip validation in dor-services-app?

**Review by ndushay — APPROVED** [review]
approved and dropped a question

File: README.md (hunk 2 above)
Comment (mjgiarlo) [reply README.md:51]: I'd rather not get into the reasons why, if I could avoid it!

**Pull request merged**